src/QuadForm/Herm/Spaces.jl

```diff
@@ -5,26 +5,26 @@
 ################################################################################
 
 @doc Markdown.doc"""
-    hermitian_space(K::NumField, n::Int; cached = true) -> HermSpace
+    hermitian_space(E::NumField, n::Int; cached = true) -> HermSpace
 
-Create the Hermitian space over `K` with dimension `n` and Gram matrix equal to
-the identity matrix. The number field `K` must be a quadratic extension, that
-is, `degree(K) == 2` must hold.
+Create the Hermitian space over `E` with dimension `n` and Gram matrix equals to
+the identity matrix. The number field `E` must be a quadratic extension, that
+is, `degree(E) == 2` must hold.
 """
-function hermitian_space(K::NumField, n::Int; cached::Bool = true)
-  G = identity_matrix(K, n)
-  return hermitian_space(K, G, cached = cached)
+function hermitian_space(E::NumField, n::Int; cached::Bool = true)
+  G = identity_matrix(E, n)
+  return hermitian_space(E, G, cached = cached)
 end
 
 @doc Markdown.doc"""
-    hermitian_space(E::NumField, G::MatElem; cached = true) -> HermSpace
+    hermitian_space(E::NumField, gram::MatElem; cached = true) -> HermSpace
 
-Create the Hermitian space over `K` with Gram matrix equal to the identity
-matrix. The matrix `G` must be square and Hermitian with respect to the non-trivial
-automorphism of `K`. The number field `K` must be a quadratic extension, that
-is, `degree(K) == 2` must hold.
+Create the Hermitian space over `E` with Gram matrix equals to `gram`. The matrix `gram` 
+must be square and Hermitian with respect to the non-trivial automorphism of `E`. 
+The number field `E` must be a quadratic extension, that is, `degree(E) == 2` must hold.
 """
 function hermitian_space(E::NumField, gram::MatElem; cached::Bool = true)
+  @req (E isa NfRel && degree(E) == 2) "E must be a relative number field of degree 2"
```

Comment:
No, this is not necessary.
```julia
julia> K, = quadratic_field(-1)
(Imaginary quadratic field defined by x^2 + 1, sqrt(-1))

julia> V = hermitian_space(K, 2)
Hermitian space over
Imaginary quadratic field defined by x^2 + 1
with Gram matrix
[1 0; 0 1]

julia> V = hermitian_space(K, K[1 2; 2 1])
Hermitian space over
Imaginary quadratic field defined by x^2 + 1
with Gram matrix
[1 2; 2 1]
```

Reply:
Oh ok, my guess here was that we really want that the fixed field is of `NumField` type, so that we have to consider `QQ` has the trivial number field. So here we only need the degree to be 2 ?

Reply:
Yes, I think we should restrict to degree `2` and this will be enough.